src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc

```diff
@@ -0,0 +1,220 @@
+ public:
+  SubscribeVehicleDataRequestTest()
+      : app_(CreateMockApp())
+      , mock_message_helper_(*am::MockMessageHelper::message_helper_mock())
+
+  {
+    msg_ = CreateMessage(smart_objects::SmartType_Map);
+    data_type = am::VehicleDataType::SPEED;
+    vehicle_data.insert(
+        std::pair<std::string, am::VehicleDataType>(kKeyName, data_type));
+    EXPECT_CALL(*application_manager::MockMessageHelper::message_helper_mock(),
+                vehicle_data()).WillRepeatedly(ReturnRef(vehicle_data));
+  }
+
+  ~SubscribeVehicleDataRequestTest() {
+    // Fix DataAccessor release and WinQt crash
+    Mock::VerifyAndClearExpectations(&app_mngr_);
+  }
+
+ protected:
+  MockAppPtr app_;
+  //  MessageSharedPtr command_msg_;
+  MessageSharedPtr msg_;
+  ::utils::SharedPtr<SubscribeVehicleDataRequest> command_;
+  am::MockMessageHelper& mock_message_helper_;
+
+  am::VehicleData vehicle_data;
+  am::VehicleDataType data_type;
+  sync_primitives::Lock app_set_lock_;
+};
+
+class TestSubscribeVehicleDataRequest : public SubscribeVehicleDataRequest {
+ public:
+  TestSubscribeVehicleDataRequest(const MessageSharedPtr& message,
+                                  am::ApplicationManager& application_manager)
+      : SubscribeVehicleDataRequest(message, application_manager) {}
+
+  am::CommandParametersPermissions& removed_parameters_permissions() {
+    return removed_parameters_permissions_;
+  }
+};
+
+TEST_F(SubscribeVehicleDataRequestTest, Run_DataAlreadySubscribed_IGNORED) {
+  (*msg_)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  (*msg_)[am::strings::msg_params][kKeyName] = true;
+  CommandPtr command(CreateCommand<SubscribeVehicleDataRequest>(msg_));
+
+  ON_CALL(app_mngr_, application(kConnectionKey)).WillByDefault(Return(app_));
+  // Data is subscribed
+  EXPECT_CALL(*app_, IsSubscribedToIVI(data_type)).WillOnce(Return(true));
+
+  const std::string info = "Already subscribed on some provided VehicleData.";
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(
+                  MobileResponseIs(mobile_result::IGNORED, info, true), _));
+
+  command->Run();
+}
+
+TEST_F(SubscribeVehicleDataRequestTest,
+       OnEvent_HMIRespondSuccess_DataSubscribed) {
+  MockAppPtr mock_app(CreateMockApp());
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app));
+
+  (*msg_)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  (*msg_)[am::strings::msg_params][kKeyName] = true;
+  CommandPtr command(CreateCommand<SubscribeVehicleDataRequest>(msg_));
+
+  // Data previously is not subscribed
+  EXPECT_CALL(*mock_app, IsSubscribedToIVI(_)).WillRepeatedly(Return(false));
+
+  am::ApplicationSet app_set;
+  app_set.insert(mock_app);
+
+  DataAccessor<am::ApplicationSet> accessor(app_set, app_set_lock_);
+  EXPECT_CALL(app_mngr_, applications()).WillOnce(Return(accessor));
+
+  // Command was sent to HMI
+  EXPECT_CALL(app_mngr_, ManageHMICommand(_));
```

Comment:
@LevchenkoS In this case - yes. This test check will app subscribe after HMI respond or not, so check here message that will be sent on HMI is not necessary